Wikipedia/Code/WMFArticleFetcher.m

```diff
@@ -173,6 +173,10 @@ - (id)serializedArticleWithURL:(NSURL *)url response:(NSDictionary *)response {
     MWKArticle *article = [[MWKArticle alloc] initWithURL:url dataStore:self.dataStore];
     @try {
         [article importMobileViewJSON:response];
+        if([NSLocale wmf_isCurrentLocaleChineseVariant]){
```

Comment:
Instead of checking the locale, this should check the article language. A user  with an English locale can have a preferred order of Chinese language variants in settings that would affect how zh-wiki articles are loaded.